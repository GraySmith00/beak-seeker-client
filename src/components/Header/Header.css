.header {
  display: flex;
  justify-content: space-around;
  align-items: center;
  position: fixed;
  top: 0;
  height: 50px;
  width: 375px;
  background: crimson; }
  .header .back-icon {
    position: absolute;
    left: 20px;
    font-size: 1.8rem;
    cursor: pointer; }
