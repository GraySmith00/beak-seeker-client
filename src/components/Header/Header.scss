.header {
  display: flex;
  justify-content: space-around;
  align-items: center;

  position: fixed;
  top: 0;
  height: 60px;
  width: 375px;

  background: #1b2026;
  color: #fff;

  z-index: 99;

  .back-icon {
    position: absolute;
    left: 20px;
    font-size: 1.8rem;
    cursor: pointer;
  }
}
