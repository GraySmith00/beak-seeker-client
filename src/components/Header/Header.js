import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';

import './Header.css';

const Header = ({ currentPage, history }) => {
  return (
    <div className="header">
      <span onClick={history.goBack} className="back-icon">
        <i className="fas fa-angle-left" />
      </span>
      <h3>{currentPage}</h3>
    </div>
  );
};

Header.propTypes = {
  currentPage: PropTypes.string.isRequired,
  history: PropTypes.object.isRequired
};

export default withRouter(Header);
