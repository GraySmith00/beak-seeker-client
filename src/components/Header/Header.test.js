import React from 'react';
import { shallow } from 'enzyme';

import { Header } from './Header';

describe('Header component', () => {
  let wrapper;
  let mockHistory;
  let mockFn;

  beforeEach(() => {
    mockFn = jest.fn();
    mockHistory = {
      goBack: mockFn
    };
    wrapper = shallow(<Header currentPage={'Home'} history={mockHistory} />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the goBack method on history when clicked', () => {
    wrapper.find('.back-icon').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});
