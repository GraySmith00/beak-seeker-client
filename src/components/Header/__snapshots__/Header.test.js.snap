// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentPage="Home"
    history={
        Object {
            "goBack": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="back-icon"
          onClick={[Function]}
>
          <i
                    className="fas fa-angle-left"
          />
</span>,
        <h3>
          <span>
                    <i
                              className="fas fa-home"
                    />
                     Home
          </span>
</h3>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-angle-left"
/>,
          "className": "back-icon",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-angle-left",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            <i
                        className="fas fa-home"
            />
             Home
</span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="fas fa-home"
/>,
              " Home",
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-home",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            " Home",
          ],
          "type": "span",
        },
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="back-icon"
            onClick={[Function]}
>
            <i
                        className="fas fa-angle-left"
            />
</span>,
          <h3>
            <span>
                        <i
                                    className="fas fa-home"
                        />
                         Home
            </span>
</h3>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="fas fa-angle-left"
/>,
            "className": "back-icon",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-angle-left",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              <i
                            className="fas fa-home"
              />
               Home
</span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="fas fa-home"
/>,
                " Home",
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-home",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              " Home",
            ],
            "type": "span",
          },
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
